src/Sarif.Viewer.VisualStudio/Models/CallTree.cs

```diff
@@ -37,34 +42,184 @@ public ObservableCollection<CallTreeNode> TopLevelNodes
             }
         }
 
-        public CodeLocationObject SelectedItem
+
+        public CallTreeNode SelectedItem
         {
             get
             {
-                return _selectedItem;
+                return this._selectedItem;
             }
             set
             {
+                this._selectedItem = value;
+                this.NotifyPropertyChanged(nameof(SelectedItem));
+                
                 // Remove the existing highlighting.
                 if (_selectedItem != null)
                 {
                     _selectedItem.ApplyDefaultSourceFileHighlighting();
                 }
+            }
+        }
 
-                _selectedItem = value;
-                this.NotifyPropertyChanged(nameof(SelectedItem));
+        private static int GetIndexInCallTreeNodeList(IList<CallTreeNode> list, CallTreeNode givenNode)
+        {
+            int index = 0;
+            foreach (CallTreeNode listNode in list)
+            {
+                if (listNode == givenNode)
+                {
+                    return index;
+                }
+                else
+                {
+                    index++;
+                }
+            }
 
-                // Navigate to the source of the selected node and highlight the region.
-                if (_selectedItem != null)
+            /* this exception should never be thrown, as this method should only be called when 
+             * givenNode is already known to be a member of list 
+             */
+            throw new System.Exception("Given node was not in the list.");
```

Comment:
>Exception [](start = 29, length = 9)

You should probably throw something more specific than System.Exception (so your callers don't need to catch System.Exception.) Maybe an IndexOutOfRangeException?